examples/nami/Pkh2Pkh.purs

```diff
@@ -0,0 +1,149 @@
+--
+-- This module demonstrates how the `QueryM` interface can be used to build a
+-- transaction from scratch. It creates and balances an example transaction that
+-- gets UTxOs from the user's wallet and sends two Ada back to the same wallet
+-- address
+--
+-- * Prerequisites
+--   - A Chromium-based browser
+--
+--   - A Nami wallet funded with test Ada ("tAda") and collateral set, If you need
+--     tAda, visit https://testnets.cardano.org/en/testnets/cardano/tools/faucet/
+--
+-- * How to run
+--
+--   The `QueryM` interface requires several external services to be running. From
+--   the repository root, run the following commands:
+--
+--   - `make run-tesnet-node`
+--      Starts a testnet Cardano node. May take some time to sync fully
+--
+--   - `make run-tesnet-ogmios`
+--      Starts the Ogmios service. Also needs to sync with the running node
+--
+--   - `make run-haskell-server`
+--      Starts the external Haskell server that will perform the transaction
+--      fee calculations (no sync required)
+--
+--   Once these services are *fully synced*, run:
+--
+--   - `npm run dev` and visit `localhost:4008`. You may be prompted to enable
+--     access to your wallet if you have not run this example before. You will
+--     also be prompted to sign the transaction using your Nami password
+
+module Examples.Nami.Pkh2Pkh (main) where
+
+import Prelude
+
+import BalanceTx (balanceTxM)
+import Control.Monad.Error.Class (throwError)
+import Control.Monad.Reader (runReaderT)
+import Data.Array as Array
+import Data.BigInt as BigInt
+import Data.Either (either)
+import Data.Map as Map
+import Data.Maybe (Maybe(Just, Nothing), maybe)
+import Data.Newtype (unwrap)
+import Data.Tuple (fst)
+import Effect (Effect)
+import Effect.Class (liftEffect)
+import Effect.Aff (error, launchAff_)
+import Effect.Aff.Class (liftAff)
+import Effect.Console as Console
+import QueryM
+  ( QueryM
+  , defaultServerConfig
+  , getWalletAddress
+  , mkOgmiosWebSocketAff
+  , utxosAt
+  )
+import Serialization as Serialization
+import Types.ByteArray (byteArrayToHex)
+import Types.POSIXTimeRange
+  ( Extended(NegInf, PosInf)
+  , Interval(Interval)
+  , LowerBound(LowerBound)
+  , UpperBound(UpperBound)
+  )
+import Types.Transaction
+  ( NetworkId(Testnet)
+  , Transaction(Transaction)
+  , TransactionOutput(TransactionOutput)
+  , TxBody(TxBody)
+  , emptyTransactionWitnessSet
+  )
+import Types.UnbalancedTransaction (UnbalancedTx(UnbalancedTx))
+import Types.Value as Value
+import Untagged.Union (asOneOf)
+import Wallet (mkNamiWalletAff)
+
+main :: Effect Unit
+main = launchAff_ $ do
+  wallet <- Just <$> mkNamiWalletAff
+  ws <- mkOgmiosWebSocketAff "ws:127.0.0.1:1337"
```

Comment:
Is it worth centralising this websocket to a string somewhere, it's also used in tests. Not actually sure how often it will appear in the future admittedly.

Reply:
The `mkOgmiosWebSocketAff` will be overhauled in the future (see #95)